src/main/java/org/kainos/ea/db/AuthDao.java

```diff
@@ -0,0 +1,111 @@
+            String SQL = "SELECT Password FROM `User` WHERE Username = ?";
+            PreparedStatement st = conn.prepareStatement(SQL);
+
+            st.setString(1,credential.getUsername());
+
+            ResultSet rs = st.executeQuery();
+
+            while(rs.next()) {
+                return rs.getString("Password").equals(credential.getPassword());
+            }
+        } catch (SQLException e) {
+            System.err.println(e.getMessage());
+            return false;
+        }
+
+        return false;
+    }
+
+    @Override
+    public boolean registerUser(Credential credential) throws GenericActionFailedException, AuthenticationException {
+        try {
+            Connection conn = DatabaseConnector.getConnection();
+            System.out.println(credential.getUsername() + " " + credential.getPassword());
+            String SQL = "INSERT INTO User(Username,Password,RoleID) VALUES(?,?,2)";
+            PreparedStatement st = conn.prepareStatement(SQL, Statement.RETURN_GENERATED_KEYS);
+
+            st.setString(1,credential.getUsername());
+            st.setString(2,credential.getPassword());
+
+            st.executeUpdate();
+            ResultSet rs = st.getGeneratedKeys();
+
+            if(rs.next()){
+                return true;
+            }
+
+            throw new AuthenticationException("Failed to register user");
+        } catch (SQLException e) {
+            throw new GenericActionFailedException(e.getMessage());
+        }
+    }
+
+    @Override
+    public String generateToken(String username) throws GenericActionFailedException {
+        try {
+            String token = UUID.randomUUID().toString();
+            Date expiry = DateUtils.addHours(new Date(),8);
+
+            Connection c = DatabaseConnector.getConnection();
+
+            String insertStatement = "INSERT INTO Token(Username,Token, Expiry) VALUES(?,?,?)";
+
+            PreparedStatement st = c.prepareStatement(insertStatement);
+
+            st.setString(1,username);
+            st.setString(2,token);
+            st.setTimestamp(3,new java.sql.Timestamp(expiry.getTime()));
+
+            st.executeUpdate();
+
+            return token;
+        } catch (SQLException e){
+            throw new GenericActionFailedException(e.getMessage());
+        }
+    }
+
+    @Override
+    public UserRole getRoleIdForToken(String token) throws GenericActionFailedException,AuthenticationException {
+        try{
+            Connection conn = DatabaseConnector.getConnection();
+            String SQL = "SELECT RoleID, Expiry FROM `User` JOIN `Token` using (Username) WHERE Token = ?";
+
+            PreparedStatement st = conn.prepareStatement(SQL);
+
+            st.setString(1,token);
+            ResultSet rs = st.executeQuery();
+
+            while(rs.next()) {
+                Timestamp expiry = rs.getTimestamp("Expiry");
+
+                if(expiry.after(new Date())) {
+                    return UserRole.values()[rs.getInt(1) - 1];
```

Comment:
can you split this over 2 lines with comments to improve clarity

Reply:
Yeah two lines is a lot clearer. Added a comment here just to be sure